src/plugins/ImportKeras/ImportKeras.js

```diff
@@ -0,0 +1,299 @@
+ * A plugin that inherits from the PluginBase. To see source code documentation about available
+ * properties and methods visit %host%/docs/source/PluginBase.html.
+ */
+
+define([
+    'plugin/PluginConfig',
+    'text!./metadata.json',
+    'plugin/PluginBase',
+    './utils/JSONModelMaps',
+    './utils/json-model-parser',
+], function (
+    PluginConfig,
+    pluginMetadata,
+    PluginBase,
+    ModelMaps,
+    JSONLayerParser,
+) {
+    'use strict';
+
+    pluginMetadata = JSON.parse(pluginMetadata);
+
+
+    /**
+     * Initializes a new instance of ImportKeras.
+     * @class
+     * @augments {PluginBase}
+     * @classdesc This class represents the plugin ImportKeras.
+     * @constructor
+     */
+    var ImportKeras = function () {
+        // Call base class' constructor.
+        PluginBase.call(this);
+        this.pluginMetadata = pluginMetadata;
+    };
+
+    /**
+     * Metadata associated with the plugin. Contains id, name, version, description, icon, configStructure etc.
+     * This is also available at the instance at this.pluginMetadata.
+     * @type {object}
+     */
+    ImportKeras.metadata = pluginMetadata;
+
+    // Prototypical inheritance from PluginBase.
+    ImportKeras.prototype = Object.create(PluginBase.prototype);
+    ImportKeras.prototype.constructor = ImportKeras;
+
+    /**
+     * Main function for the plugin to execute. This will perform the execution.
+     * Notes:
+     * - Always log with the provided logger.[error,warning,info,debug].
+     * - Do NOT put any user interaction logic UI, etc. inside this method.
+     * - callback always has to be called even if error happened.
+     *
+     * @param {function(Error|null, plugin.PluginResult)} callback - the result callback
+     */
+    ImportKeras.prototype.main = async function(callback){
+        let srcJsonHash = this.getCurrentConfig().srcModel;
+        if (!srcJsonHash) {
+            callback(new Error('Keras Json Not Provided'), this.result);
+            return;
+        }
+        try {
+            this.archName = this.getCurrentConfig().archName;
+            let metadata = await this.blobClient.getMetadata(srcJsonHash);
+            this.archName = this.archName ? this.archName : metadata.name.replace('.json', '');
+            let modelJson = await this.blobClient.getObjectAsJSON(srcJsonHash);
+            this.modelInfo = JSONLayerParser.flatten(modelJson).config;
+            this.addNewArchitecture();
+            this.addLayers();
+            await this.addConnections();
+            await this.save('Completed Import Model');
+            this.result.setSuccess(true);
+            callback(null, this.result);
+        } catch (err) {
+            this.logger.debug(`Something Went Wrong, Error Message: ${err}`);
+            callback(err, this.result);
+        }
+    };
+
+    ImportKeras.prototype.addNewArchitecture = function () {
+        // Add Architecture
+        this.importedArch = this.core.createNode({
```

Comment:
Similar to my earlier comment, it might make more sense to return this node and then pass it to `addLayers`. It is a bit easier to follow rather than making it accessible to every function in the plugin.